contracts/BC_fusion/StakeHub.sol

```diff
@@ -0,0 +1,672 @@
+        require(newOperator != address(0), "INVALID_OPERATOR");
+        Validator storage valInfo = validators[validator];
+        require(valInfo.updateTime + 1 days <= block.timestamp, "UPDATE_TOO_FREQUENTLY");
+        valInfo.operatorAddress = newOperator;
+        valInfo.updateTime = block.timestamp;
+
+        emit OperatorEdited(validator, newOperator);
+    }
+
+    function editDescription(address validator, Description calldata description) external onlyInitialized onlyOperator(validator) {
+        Validator storage valInfo = validators[msg.sender];
+
+        valInfo.description = description;
+        valInfo.updateTime = block.timestamp;
+
+        emit DescriptionEdited(validator);
+    }
+
+    function editVoteAddress(address validator, bytes calldata newVoteAddress, bytes calldata blsProof) external onlyInitialized onlyOperator(validator) {
+        Validator storage valInfo = validators[msg.sender];
+        require(valInfo.updateTime + 1 days <= block.timestamp, "UPDATE_TOO_FREQUENTLY");
+        require(voteAddressToValidator[newVoteAddress] == address(0), "DUPLICATE_VOTE_ADDRESS");
+        require(_checkVoteAddress(newVoteAddress, blsProof), "INVALID_VOTE_ADDRESS");
+        voteAddressToValidator[newVoteAddress] = msg.sender;
+        valInfo.voteAddress = newVoteAddress;
+        valInfo.updateTime = block.timestamp;
+
+        emit VoteAddressEdited(validator, newVoteAddress);
+    }
+
+    function editCommissionRate(address validator, uint256 commissionRate) external onlyInitialized validatorExist(validator) {
+        Validator storage valInfo = validators[msg.sender];
+        require(valInfo.updateTime + 1 days <= block.timestamp, "UPDATE_TOO_FREQUENTLY");
+        require(commissionRate <= valInfo.commission.maxRate, "INVALID_COMMISSION_RATE");
+
+        if (commissionRate > valInfo.commission.rate) {
+            require(commissionRate - valInfo.commission.rate <= valInfo.commission.maxChangeRate, "INVALID_MAX_CHANGE_RATE");
+        } else {
+            require(valInfo.commission.rate - commissionRate <= valInfo.commission.maxChangeRate, "INVALID_MAX_CHANGE_RATE");
+        }
+
+        valInfo.commission.rate = commissionRate;
+        valInfo.updateTime = block.timestamp;
+
+        emit CommissionRateEdited(validator, commissionRate);
+    }
+
+    function unjail(address validator) public onlyInitialized validatorExist(validator) {
+        Validator storage valInfo = validators[validator];
+        require(valInfo.jailed, "NOT_JAILED");
+
+        address pool = valInfo.poolModule;
+        require(IStakePool(pool).getSecurityDepositBNB() >= minSelfDelegationBNB, "NOT_ENOUGH_SELF_DELEGATION");
+        require(valInfo.jailUntil <= block.timestamp, "STILL_JAILED");
+
+        valInfo.jailed = false;
+
+        emit ValidatorUnjailed(validator);
+    }
+
+    function delegate(address validator) external payable onlyInitialized validatorExist(validator) validatorNotJailed(validator) whenNotPaused {
```

Comment:
if we do not allow delegation if a validator is in jail, how an operator of a validator to delegate more and transfer stBNB to stakehub as `SecurityDepositBNB`?

Reply:
Fixed